packages/core/src/application.ts

```diff
@@ -3,17 +3,31 @@
 // This file is licensed under the MIT License.
 // License text available at https://opensource.org/licenses/MIT
 
-import {Context, Binding, BindingScope, Constructor} from '@loopback/context';
-import {Server} from './server';
+import {
+  Binding,
+  BindingScope,
+  BindingType,
+  Constructor,
+  Context,
+} from '@loopback/context';
 import {Component, mountComponent} from './component';
 import {CoreBindings} from './keys';
+import {
+  asLifeCycleObserverBinding,
+  isLifeCycleObserver,
+  isLifeCycleObserverClass,
+  LifeCycleObserver,
+} from './lifecycle';
+import {Server} from './server';
+import debugModule = require('debug');
```

Comment:
In other places, I was already using `debugFactory` instead of `debugModule`. Let's pick one consistent way going forward please.